src/state/slices/time_slice.ts

```diff
@@ -99,3 +100,20 @@ export const addTimeDerivedStateSubscribers = (store: SubscribableStore<DatasetS
     }
   );
 };
+
+export const serializeTimeSlice = (state: TimeSlice): SerializedStoreData => {
+  return {
+    [UrlParam.TIME]: state.currentFrame.toString(),
+  };
+};
+
+export const loadTimeSliceFromParams = (state: TimeSlice & DatasetSlice, params: URLSearchParams): void => {
+  // Load time from URL. If no time is set but a track is, set the time to the
+  // start of the track.
+  const time = decodeInt(params.get(UrlParam.TIME));
+  if (time !== undefined) {
+    state.setFrame(time);
+  } else if (state.track !== null) {
+    state.setFrame(state.track.startTime());
+  }
```

Comment:
you have have a check for `isTabType` up above, but it seems to be the only param you're verifying against the current data. What if the time from the url is out of the range of the dataset?

Reply:
Good catch, I noticed this too a few days ago. I opened a fix for this as a follow-up PR, #597, which does a check for valid time values and clamps time inputs.